src/main/java/frc/robot/subsystems/scoring/AimerIOSim.java

```diff
@@ -0,0 +1,142 @@
+            new TrapezoidProfile(
+                    new TrapezoidProfile.Constraints(
+                            ScoringConstants.aimCruiseVelocity, ScoringConstants.aimAcceleration));
+
+    private final Timer timer = new Timer();
+
+    private boolean override = false;
+    private double overrideVolts = 0.0;
+
+    boolean newProfile = false;
+    double previousGoalAngle = 0.0;
+
+    double minAngleClamp = 0.0;
+    double maxAngleClamp = 0.0;
+
+    double goalAngleRad = 0.0;
+    double appliedVolts = 0.0;
+
+    double initialAngle = 0.0;
+    double initialVelocity = 0.0;
+
+    @Override
+    public void setAimAngleRad(double goalAngleRad, boolean newProfile) {
+        this.goalAngleRad = goalAngleRad;
+        this.newProfile = newProfile;
+    }
+
+    @Override
+    public void controlAimAngleRad() {
+        if (goalAngleRad != previousGoalAngle && newProfile) {
+            timer.reset();
+            timer.start();
+
+            initialAngle = sim.getAngleRads();
+            initialVelocity = sim.getVelocityRadPerSec();
+
+            previousGoalAngle = goalAngleRad;
+        }
+        goalAngleRad = MathUtil.clamp(goalAngleRad, minAngleClamp, maxAngleClamp);
+    }
+
+    @Override
+    public void setAngleClampsRad(double minAngleClamp, double maxAngleClamp) {
+        if (minAngleClamp > maxAngleClamp) {
+            return;
+        }
+        this.minAngleClamp =
+                MathUtil.clamp(minAngleClamp, 0.0, ScoringConstants.aimMaxAngleRadians);
+        this.maxAngleClamp =
+                MathUtil.clamp(maxAngleClamp, 0.0, ScoringConstants.aimMaxAngleRadians);
+    }
+
+    @Override
+    public void setOverrideMode(boolean override) {
+        this.override = override;
+    }
+
+    @Override
+    public void setOverrideVolts(double volts) {
+        appliedVolts = volts;
+    }
+
+    @Override
+    public void setPID(double p, double i, double d) {
+        controller.setP(p);
+        controller.setI(i);
+        controller.setD(d);
+    }
+
+    @Override
+    public void updateInputs(AimerIOInputs inputs) {
+        sim.update(Constants.loopTime);
+
+        State trapezoidSetpoint =
+                profile.calculate(
+                        timer.get(),
+                        new State(initialAngle, initialVelocity),
+                        new State(goalAngleRad, 0));
+
+        double controlSetpoint =
+                MathUtil.clamp(
+                        trapezoidSetpoint.position, 0.0, ScoringConstants.aimMaxAngleRadians);
+        double velocitySetpoint = trapezoidSetpoint.velocity;
+
+        if (override) {
+            appliedVolts = overrideVolts;
+        } else {
+            appliedVolts =
+                    feedforward.calculate(controlSetpoint, velocitySetpoint)
+                            + controller.calculate(sim.getAngleRads(), controlSetpoint);
+            appliedVolts = MathUtil.clamp(appliedVolts, -12.0, 12.0);
+        }
```

Comment:
similar comments for the other IO class - we should have this in `applyOutputs`

Also, if this code is repetitive, we should find a way to put it in a separate file and just use it once. It's better to not maintain two copies of the same code.